lib/dora-parser/src/error/msg.rs

```diff
@@ -369,22 +369,23 @@ impl Msg {
 
 #[derive(Clone, Debug)]
 pub struct MsgWithPos {
-    pub msg: Msg,
+    pub file: String,
     pub pos: Position,
+    pub msg: Msg,
 }
 
 impl MsgWithPos {
-    pub fn new(pos: Position, msg: Msg) -> MsgWithPos {
-        MsgWithPos { pos: pos, msg: msg }
+    pub fn new(file: String, pos: Position, msg: Msg,) -> MsgWithPos {
+        MsgWithPos { file: file, pos: pos, msg: msg }
     }
 
     pub fn message(&self) -> String {
-        format!("error at {}: {}", self.pos, self.msg.message())
+        format!("error1 in {} at {}: {}", self.file, self.pos, self.msg.message())
     }
 }
 
 impl fmt::Display for MsgWithPos {
     fn fmt(&self, f: &mut fmt::Formatter) -> Result<(), fmt::Error> {
-        write!(f, "error at {}: {}", self.pos, self.msg.message())
+        write!(f, "error2 in {} at {}: {}", self.file, self.pos, self.msg.message())
```

Comment:
I think we should make this function something like:

```rust
if let Some(file) = self.file {
    write!(f, "error in {} at {}: {}", self.file, self.pos, self.msg.message())
} else {
    write!(f, "error at {}: {}", self.pos, self.msg.message())
}
```

In our Ruby script we can then just parse both error formats.